kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/service/impl/B2BPurchaseOrderSciquestServiceImpl.java

```diff
@@ -0,0 +1,375 @@
+                    cXml.append("        <ProductReferenceNumber><![CDATA[" + bn(poi.getExternalOrganizationB2bProductReferenceNumber()) + "]]></ProductReferenceNumber>\n");
+                }
+                // ProductType - Describes the type of the product or service. Valid values: Catalog, Form, Punchout. Mandatory.
+                cXml.append("        <ProductType>").append(poi.getExternalOrganizationB2bProductTypeName()).append("</ProductType>\n");
+                cXml.append("      </Item>\n");
+                cXml.append("      <Quantity>").append(poi.getItemQuantity()).append("</Quantity>\n");
+
+                String taxable = (poi.getItemType().isTaxableIndicator() ? "true" : "false");
+                cXml.append("      <Taxable>" + taxable + "</Taxable>\n");
+
+                // LineCharges - All the monetary charges for this line, including the price, tax, shipping, and handling.
+                // Required.
+                cXml.append("      <LineCharges>\n");
+                cXml.append("        <UnitPrice>\n");
+                cXml.append("          <Money currency=\"USD\">").append(poi.getItemUnitPrice()).append("</Money>\n");
+                cXml.append("        </UnitPrice>\n");
+                cXml.append("        <ExtendedPrice>\n");
+                cXml.append("          <Money currency=\"USD\">").append(poi.getExtendedPrice()).append("</Money>\n");
+                cXml.append("        </ExtendedPrice>\n");
+
+                KualiDecimal taxAmount = poi.getItemTaxAmount();
+                KualiDecimal preTaxAmount = poi.getExtendedPrice();
+                KualiDecimal totalTaxRate = calcTaxRate(preTaxAmount, taxAmount);
+
+                // only send tax information if the line item is taxable and it's sales tax. dont send use tax dollars.
+                if (taxable.equalsIgnoreCase("true") && !purchaseOrder.isUseTaxIndicator()) {
+                    cXml.append("        <Tax1>\n");
+                    cXml.append("          <TaxShippingHandling>\n");
+                    cXml.append("            <TSHConfig override=\"true\">\n");
+                    cXml.append("              <FlatChargeByLine>\n");
+                    cXml.append("                <Money currency=\"USD\">" + (taxable.equalsIgnoreCase("false") || taxAmount == null ? "0.00" : taxAmount.toString()) + "</Money>\n");
+                    cXml.append("              </FlatChargeByLine>\n");
+                    cXml.append("            </TSHConfig>\n");
+                    cXml.append("          </TaxShippingHandling>\n");
+                    cXml.append("        </Tax1>\n");
+                }
+
+                // pass in whether the POLine is taxable or not
+                cXml.append("      </LineCharges>\n");
+
+                cXml.append("    </POLine>\n");
+            }
+        }
+        return cXml;
+    }
+
+    protected KualiDecimal calcTaxRate(KualiDecimal preTaxAmount, KualiDecimal taxAmount) {

```

Comment:
This is not needed if we remove the writePOLines() and take what is in baseline KFS6 for getCxml().